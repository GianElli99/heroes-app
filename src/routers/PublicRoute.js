import React from 'react';
import { Redirect, Route, useHistory } from 'react-router-dom';
import PropTypes from 'prop-types';

export const PublicRoute = ({
  isAuthenticated,
  component: Component,
  ...rest
}) => {
  console.log(rest.location.pathname);
  return (
    <Route
      {...rest}
      component={(props) =>
        !isAuthenticated ? <Component {...props} /> : <Redirect to="/" />
      }
    />
  );
};

PublicRoute.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  component: PropTypes.func.isRequired,
};
