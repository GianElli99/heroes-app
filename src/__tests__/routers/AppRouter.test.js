import { mount, shallow } from 'enzyme';
import { AuthContext } from '../../auth/AuthContext';
import { AppRouter } from '../../routers/AppRouter';

describe('Pruebas en AppRouter', () => {
  test('should mostart el login si no estoy autenticado', () => {
    const contextValues = {
      user: {
        logged: false,
      },
      dispatch: jest.fn(),
    };

    const wrapper = mount(
      <AuthContext.Provider value={contextValues}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('debe mostar el componente marvel si esta autenticado', () => {
    const contextValues = {
      user: {
        name: 'Gian',
        logged: true,
      },
      dispatch: jest.fn(),
    };

    const wrapper = mount(
      <AuthContext.Provider value={contextValues}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper.find('.navbar').exists()).toBe(true);
  });
});
