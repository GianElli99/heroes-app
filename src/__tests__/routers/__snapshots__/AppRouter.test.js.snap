// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AppRouter should mostart el login si no estoy autenticado 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login
    </h1>
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
